### PR TITLE
Containerization of the pipeline with docker and octave

### PR DESCRIPTION
**Pull request by Remi-Gau**
The biggest change in this PR is to add the NiftiTools dependencies into the repo itself. It makes more convenient to build the docker image and to fix problems in this dependencies that are octave related.

Except from that this creates a Dockerfile and changes some lines in the the batch_download_run (see changes below).

### CODE REVIEW

File: batch.m [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
-WD = fullfile(pwd,'subfun');
+% the cdirectory with this script becomes the current directory
+WD = pwd;
+
+% we add all the subfunctions that are in the sub directories
 addpath(genpath(WD))
-toolbox_path = '/home/remi/Dropbox';
-addpath(fullfile(toolbox_path, 'Code', 'MATLAB', 'Neuroimaging', 'NiftiTools'))
+
+% In case some toolboxes need to be added the matlab path, specify and uncomment
```
Comment (Remi-Gau) [inline batch.m]: This is not needed as the dependencies are included in the repo.

File: batch.m [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
 opt = getOption();
 
-WD = fullfile(pwd,'subfun');
+% the cdirectory with this script becomes the current directory
+WD = pwd;
+
+% we add all the subfunctions that are in the sub directories
 addpath(genpath(WD))
-toolbox_path = '/home/remi/Dropbox';
-addpath(fullfile(toolbox_path, 'Code', 'MATLAB', 'Neuroimaging', 'NiftiTools'))
```
Comment (Remi-Gau) [inline batch.m]: This is not needed as the repo includes the dependencies.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# this is mostly taken from the spm docker files: https://github.com/spm/spm-docker
```
Comment (Remi-Gau) [inline Dockerfile]: This is the recipe to follow to create the docker image. Let me know if it is unclear but I tired to add a few comments here and there.

File: batch_download_run.m [hunk 4]
```diff
@@ -1,22 +1,43 @@
...
+addpath(genpath(WD)) % we add all the subfunctions that are in the sub directories
+
 
+%% Set options
 opt = getOption();
 
-opt.derivativesDir = fullfile(WD, 'MoAEpilot');
-opt.groups = {''};    % {'blnd', 'ctrl'};
-opt.subjects = {1};  % {[1:2], [1:2]};
+% respecify options here in case the getOption file has been modified on
```
Comment (Remi-Gau) [inline batch_download_run.m]: the getOption file might change often so by specifying this we make sure that this script won't fail in the future

File: getOption.m [hunk 5]
```diff
@@ -26,24 +26,21 @@
 % If  slice order is entered in time unit (ms) doing  so,  the  next  item  (Reference Slice) will contain a reference time (in
 % ms) instead of the slice index of the reference slice.
 
+opt.sliceOrder = []; % TO BE USED ONLY IF SPM_BIDS CAN'T EXTRACT SLICE INFORMATION
```
Comment (Remi-Gau) [inline getOption.m]: just some reordering of the lines

**Comment by marcobarilari** [comment]
only one thing also present on the other codes, the last output on the command window is the elapsed time but it does not match the real one. For example in this docker version is 3 secs and the matlab one is around 1 sec.

**Pull request merged**